src/Language/PureScript/CST/Convert.hs

```diff
@@ -491,7 +496,12 @@ convertDeclaration fileName decl = case decl of
       instTy
         | isJust new = AST.NewtypeInstance
         | otherwise = AST.DerivedInstance
-    pure $ AST.TypeInstanceDeclaration ann chainId 0 name'
+      clsAnn = uncurry (sourceAnnCommented fileName) $
+        if null args then
+          qualRange cls
+        else
+          (fst $ qualRange cls, snd $ typeRange $ last args)
```

Comment:
Since the same code is written twice, should that be refactored out to a separate function?